### PR TITLE
Rulesets discovery from all expected directories

### PR DESCRIPTION
**Pull request by SkullTech**
Solves #106

### CODE REVIEW

**Comment by SkullTech** [comment]
I created this branch off `yaml_rulesets` assuming that will get merged through PR #168   
If this gets approved I can rebase the commits on top of master.

**Comment by TomasTomecek** [comment]
please rebase

**Comment by SkullTech** [comment]
@lachmanfrantisek Yeah you're right! Those would very nice features to have. I'll start working on it.

**Comment by phracek** [comment]
[test]

**Comment by SkullTech** [comment]
Need a stylistic opinion, which one of the below would be more user-friendly // prettier?

```console
[sumit@hal9000 colin]$ python3 -m colin.cli.colin list-rulesets
default (/home/sumit/.local/share/colin/rulesets/default.json)
fedora (/home/sumit/.local/share/colin/rulesets/fedora.json)
default (./rulesets/default.json)
fedora-sample-ruleset (./rulesets/fedora-sample-ruleset.json)
```
```console
[sumit@hal9000 colin]$ python3 -m colin.cli.colin list-rulesets
default               (/home/sumit/.local/share/colin/rulesets/default.json)
fedora                (/home/sumit/.local/share/colin/rulesets/fedora.json)
default               (./rulesets/default.json)
fedora-sample-ruleset (./rulesets/fedora-sample-ruleset.json)
```

@lachmanfrantisek  your thoughts on this?

**Comment by lachmanfrantisek** [comment]
@SkullTech  The second one looks clearer to me.

**Comment by lachmanfrantisek** [comment]
Tests will be fixed in #171.

**Comment by SkullTech** [comment]
@lachmanfrantisek Awesome! BTW I'm done with the changes you requested. Please review :)

File: colin/core/ruleset/ruleset.py [hunk 1]
```diff
@@ -137,57 +137,65 @@ def get_ruleset_file(ruleset=None):
...
-                   .format(ruleset, ruleset_directory))
+                   .format(ruleset, ruleset_dirs))
     raise ColinRulesetException("Ruleset with the name '{}' cannot be found.".format(ruleset))
 
 
-def get_ruleset_directory():
+def get_ruleset_dirs():
     """
     Get the directory with ruleset files
     First directory to check:  $HOME/.local/share/colin/rulesets
```
[the remark this answers is not in the document]
Comment (lachmanfrantisek) [reply colin/core/ruleset/ruleset.py]: We have to search in the right order: subdir -> local -> global.

**Comment by lachmanfrantisek** [comment]
@SkullTech If you rebase now, you will get green tests..;-)

File: colin/core/ruleset/ruleset.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (SkullTech) [reply colin/core/ruleset/ruleset.py]: Oh shit I reversed the order completely! Also I missed the docstring. Another commit coming. @lachmanfrantisek Thanks for the head up!

**Comment by SkullTech** [comment]
All checks have passed :)

**Pull request merged**